File: app/src/main/java/ru/santaev/clipboardtranslator/viewmodel/HistoryViewModel.java
package ru.santaev.clipboardtranslator.viewmodel;


import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import java.util.List;

import ru.santaev.clipboardtranslator.TranslatorApp;
import ru.santaev.clipboardtranslator.db.AppDatabase;
import ru.santaev.clipboardtranslator.db.dao.TranslationDao;
import ru.santaev.clipboardtranslator.db.entity.Translation;
import ru.santaev.clipboardtranslator.util.NotificationHelper;
import ru.santaev.clipboardtranslator.util.RxHelper;

public class HistoryViewModel extends ViewModel {

    private TranslationDao translationDao;
    private LiveData<List<Translation>> history = new MutableLiveData<>();

    public HistoryViewModel() {
        translationDao = AppDatabase.getInstance().getTranslationDao();
        loadHistory();
    }

    public LiveData<List<Translation>> getHistory() {
        return history;
    }

    private void loadHistory(){
        history = translationDao.getTranslations();
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        translationDao = null;
        history = null;
    }

    public void onClickedItem(Translation translation) {
    }

    public void removeItem(Translation translation) {
        RxHelper.make(() -> {
            translationDao.delete(translation);
            return null;
        }, Throwable::printStackTrace, null, null);
    }
}

File: app/src/main/java/ru/santaev/clipboardtranslator/viewmodel/HistoryViewModel.kt
package ru.santaev.clipboardtranslator.viewmodel

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.arch.lifecycle.ViewModel
import com.example.santaev.domain.dto.TranslationDto
import com.example.santaev.domain.factory.UseCaseFactory
import ru.santaev.clipboardtranslator.util.RxHelper

class HistoryViewModel : ViewModel() {

    var history: LiveData<List<TranslationDto>> = MutableLiveData()
        private set

    init {
        loadHistory()
    }

    private fun loadHistory() {
        UseCaseFactory
                .instance
                .getGetHistoryUseCase()
                .execute()
                .compose(RxHelper.getFlowableTransformer())
                .subscribe({ (history as MutableLiveData).setValue(it) })
    }

    fun onClickedItem(translation: TranslationDto) {
    }

    fun removeItem(translation: TranslationDto) {
        //dataModel.removeTranslation(translation)
        //        .subscribe()
    }

    fun clearHistory() {
        //dataModel.deleteAll()
        //        .subscribe()
    }
}
